tests/reporting/test_market_unit.py

```diff
@@ -0,0 +1,527 @@
+
+# create stake token and association with market
+def set_mock_stake_token_value(localFixture, initializeMarket, payoutDesignatedNumerators, mockStakeTokenFactory):
+    payoutDesignatedHashValue = initializeMarket.derivePayoutDistributionHash(payoutDesignatedNumerators, False)
+    mockSecondStakeToken = localFixture.upload('solidity_test_helpers/MockStakeToken.sol', str(payoutDesignatedHashValue));
+    mockSecondStakeToken.setPayoutDistributionHash(payoutDesignatedHashValue)
+    mockStakeTokenFactory.setStakeToken(payoutDesignatedHashValue, mockSecondStakeToken.address)
+    assert initializeMarket.getStakeToken(payoutDesignatedNumerators, False) == mockSecondStakeToken.address
+    return mockSecondStakeToken
+
+def push_to_first_dispute(localFixture, chain, mockReputationToken, initializeMarket, mockStakeTokenFactory, mockReportingWindow, mockUniverse, constants, mockDisputeBond, mockDisputeBondFactory):
+    numTicks = 10 ** 10
+    mockReportingWindow.setEndTime(chain.head_state.timestamp + constants.DESIGNATED_REPORTING_DURATION_SECONDS())
+    chain.head_state.timestamp = initializeMarket.getDesignatedReportDisputeDueTimestamp() - 1
+    assert initializeMarket.getReportingState() == constants.DESIGNATED_DISPUTE()
+    mockDisputeStakeToken = set_mock_stake_token_value(localFixture, initializeMarket, [0, 0, 0, numTicks, 0], mockStakeTokenFactory)
+    mockDisputeStakeToken.setTotalSupply(10)
+    mockReportingWindow.setIsDisputeActive(True)
+
+    mockDesignatedDisputeBond = localFixture.upload('solidity_test_helpers/MockDisputeBond.sol', 'mockDesignatedDisputeBond');    
+    mockDisputeBondFactory.setCreateDisputeBond(mockDesignatedDisputeBond.address)
+    assert mockReportingWindow.getReputationToken() == mockReputationToken.address
+
+    assert initializeMarket.disputeDesignatedReport([0, 0, 0, numTicks,0], 100, False, sender=tester.k2)
+
+    assert initializeMarket.getReportingState() == constants.FIRST_DISPUTE()
+    mockFirstDisputeBond = localFixture.upload('solidity_test_helpers/MockDisputeBond.sol', 'mockFirstDisputeBond');
+    mockDisputeBondFactory.setCreateDisputeBond(mockFirstDisputeBond.address)
+    mockNextReportingWindow = localFixture.upload('solidity_test_helpers/MockReportingWindow.sol', 'mockNextReportingWindow');
+    mockUniverse.setNextReportingWindow(mockNextReportingWindow.address)
+
+    mockDisputeStakeToken = set_mock_stake_token_value(localFixture, initializeMarket, [0, 0, numTicks, 0, 0], mockStakeTokenFactory)
+    mockDisputeStakeToken.setTotalSupply(1000)
+
+    # execute disputeFirstReporters on market
+    assert initializeMarket.disputeFirstReporters([0, 0, numTicks, 0, 0], 50, False, sender=tester.k4)
+
+def push_to_designated_report_state(localFixture, mockUniverse, chain, initializeMarket, mockStakeToken, payoutDesignatedNumerators, mockStakeTokenFactory, mockReportingWindow, mockReputationToken, constants):
```

Comment:
Since at the end of this call the market is in designated dispute I think this should be renamed `push_to_designated_dispute_state`

Reply:
good point, will make change